### ISSUE TITLE
Setting seed for Data Samplers (torch.utils.data.sampler) 

### ISSUE THREAD

**Issue by Akella17** [pytorch/pytorch#18171]
Is it possible to assign a random seed for BatchSampler and SubsetRandomSampler? I am using a generator function (yields a generator object instead of returning it, as storing the data in memory turned out to be a lot slower). This function has to be called twice, while it is expected to return the same output.

**Comment by fmassa** [comment]
You can pass a `init_fn` in `DataLoader` for controlling this.
